src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,176 @@
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The parent resource name.")]
+        public string ParentResourceName { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub ParentResource { get; set; }
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubResourceId,
+            HelpMessage = "The parent resource id.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs")]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The remote virtual network to which this hub virtual network connection is connected.")]
+        [ResourceGroupCompleter]
+        public PSVirtualNetwork RemoteVirtualNetwork { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The remote virtual network id to which this hub virtual network connection is connected.")]
+        [ResourceGroupCompleter]
+        [ResourceIdCompleter("Microsoft.Network/virtualNetworks")]
+        public string RemoteVirtualNetworkId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Enable internet security for this connection.")]
+        public bool? EnableInternetSecurity { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            WriteWarning("The output object type of this cmdlet will be modified in a future release.");
```

Comment:
Why?

Reply:
For some reason, most of the commandlets I see have this warning. So I assumed this is required for new commandlets. I have removed them for now.

Please let me know if the requirement is otherwise.